include/ada/urlpattern_internals.h

```diff
@@ -0,0 +1,25 @@
+#ifndef ADA_URLPATTERN_INTERNALS_H
+#define ADA_URLPATTERN_INTERNALS_H
+
+#include "ada/urlpattern_base.h"
+#include <string_view>
+#include "regex"
+#include "vector"
+
+namespace ada::urlpattern {
+// https://wicg.github.io/urlpattern/#component
+struct urlpattern_component {
```

Comment:
Since this is already under ada::urlpattern namespace, why not call this struct component?